// socialvi/application.js
exports.users=users=[{
  idUser:1,
  token:1,
  requestSend:[],
  requestRecived:[],
  friends:[{id:3}],
  posts:[]
},{
idUser:2,
token:2,
requestSend:[],
requestRecived:[],
friends:[],
posts:[]
},
{
idUser:3,
token:3,
requestSend:[],
requestRecived:[],
friends:[{id:1}],
posts:[]
}
];

var token = [1,2,3];

exports.registration = function(){
  users.push({
    idUser : users[users.length-1].idUser + 1,
    token : users[users.length-1].token + 1,
    requestSend:[],
    requestRecived:[],
    friends:[],
    posts:[]
  });
  token.push(users[users.length-1].token);
};

exports.showUsers = function(){
  array=[];
  for(var i in users){
    array.push({user: users[i].idUser});
  }
  return array;
};

exports.friendShipSend = function(idSender,idConsignee){
  for(var i in users){
    if(idSender==users[i].idUser){
      users[i].requestSend.push({idConsignee:idConsignee});
    }
    if(idConsignee==users[i].idUser){
      users[i].requestRecived.push({idSender:idSender});
    }
  }
  return {message:"friendShip Sended"};
};

exports.confirmFriendShip = function(idUser,idConfirm){
  check=this.checkFriend(idUser,idConfirm);
  if(check==false){
  for(var i in users){
    if(idUser==users[i].idUser){
        users[i].friends.push({id:idConfirm});
    }
    if(idConfirm==users[i].idUser){
      users[i].friends.push({id:idUser});
    }
  }
  return {message:"FriendShip confirmed"};
}else{
  return {message:"Already Friends"};
}
};

exports.checkFriend = function(idUser,idFriend){
  check=false;
  for(var i in users){
    if(idUser==users[i].idUser){
      for(var l in users[i].friends){
        if(users[i].friends[l].id==idFriend){
          check=true;
        }
      }
    }
  }
    return check;
};

exports.addPost = function(idUser,title){
  for(var i in users){
    if(idUser==users[i].idUser){
      users[i].posts.push({
        title:title
      });
      return {message:"post added successfully"};
    }
  }
};

exports.deletePost = function(idUser,index){
  l=index-1;
  for(var i in users){
  if(idUser==users[i].idUser){
    users[i].posts.splice(l,1);
  }
}
return {message:"post deleted"};
};

exports.showPostFriend = function(idUser,idFriend){
  check=exports.checkFriend(idUser,idFriend);
  if(check==true){
    for(var i in users){
      if(idFriend==users[i].idUser)
      return users[i].posts;
    }
  }else{
    return {message:"this id isn't your friend"};
  }
};
